commit dffa6f5ecd1f8b7211a65b762e3201a318a23538 (refs/changes/29/929/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-11-21T15:22:32-08:00 (2 years, 11 months ago)
    
    joyent/node-cueball#51 pool should never call .close() on a busy backend

diff --git a/lib/pool.js b/lib/pool.js
index 9103136..5c705d0 100644
--- a/lib/pool.js
+++ b/lib/pool.js
@@ -447,7 +447,12 @@ CueBallConnectionPool.prototype._rebalance = function () {
 		 */
 		var fsmIdx = self.p_connections[fsm.cf_backend.key].
 		    indexOf(fsm);
-		if (fsm.isInState('idle') || fsmIdx > 0 ||
+		if (fsm.isInState('idle')) {
+			fsm.close();
+			--total;
+		} else if (fsm.isInState('busy')) {
+			fsm.closeAfterRelease();
+		} else if (fsmIdx > 0 ||
 		    self.p_keys.indexOf(fsm.cf_backend.key) === -1) {
 			fsm.close();
 			--total;
